diff -up ppp-ppp-2.5.0/pppd/pppd.8.1~ ppp-ppp-2.5.0/pppd/pppd.8
--- ppp-ppp-2.5.0/pppd/pppd.8.1~	2024-07-14 00:22:59.570295039 +0100
+++ ppp-ppp-2.5.0/pppd/pppd.8	2024-07-14 00:25:18.120714756 +0100
@@ -1189,7 +1189,7 @@ Ask the peer for up to 2 DNS server addr
 by the peer (if any) are passed to the /etc/ppp/ip\-up script in the
 environment variables DNS1 and DNS2, and the environment variable
 USEPEERDNS will be set to 1.  In addition, pppd will create an
-/etc/ppp/resolv.conf file containing one or two nameserver lines with
+/run/ppp/resolv.conf file containing one or two nameserver lines with
 the address(es) supplied by the peer.
 .TP
 .B usepeerwins
diff -up ppp-ppp-2.5.0/scripts/ip-down.local.add.1~ ppp-ppp-2.5.0/scripts/ip-down.local.add
--- ppp-ppp-2.5.0/scripts/ip-down.local.add.1~	2023-03-18 09:11:05.000000000 +0000
+++ ppp-ppp-2.5.0/scripts/ip-down.local.add	2024-07-14 00:22:59.570295039 +0100
@@ -9,12 +9,13 @@
 #
 # Nick Walker (nickwalker@email.com)
 #
+. /etc/sysconfig/network-scripts/network-functions
 
-if [ -n "$USEPEERDNS" -a -f /etc/ppp/resolv.conf ]; then
-	if [ -f /etc/ppp/resolv.prev ]; then
-		cp -f /etc/ppp/resolv.prev /etc/resolv.conf
+if [ -n "$USEPEERDNS" -a -f /var/run/ppp/resolv.conf ]; then
+	if [ -f /var/run/ppp/resolv.prev ]; then
+		change_resolv_conf /var/run/ppp/resolv.prev
 	else
-		rm -f /etc/resolv.conf
+		change_resolv_conf
 	fi
 fi
 
diff -up ppp-ppp-2.5.0/scripts/ip-up.local.add.1~ ppp-ppp-2.5.0/scripts/ip-up.local.add
--- ppp-ppp-2.5.0/scripts/ip-up.local.add.1~	2023-03-18 09:11:05.000000000 +0000
+++ ppp-ppp-2.5.0/scripts/ip-up.local.add	2024-07-14 00:22:59.570295039 +0100
@@ -9,16 +9,19 @@
 #
 # Nick Walker (nickwalker@email.com)
 #
+. /etc/sysconfig/network-scripts/network-functions
 
-if [ -n "$USEPEERDNS" -a -f /etc/ppp/resolv.conf ]; then
-	rm -f /etc/ppp/resolv.prev
+if [ -n "$USEPEERDNS" -a -f /var/run/ppp/resolv.conf ]; then
+	rm -f /var/run/ppp/resolv.prev
 	if [ -f /etc/resolv.conf ]; then
-		cp /etc/resolv.conf /etc/ppp/resolv.prev
-		grep domain /etc/ppp/resolv.prev > /etc/resolv.conf
-		grep search /etc/ppp/resolv.prev >> /etc/resolv.conf
-		cat /etc/ppp/resolv.conf >> /etc/resolv.conf
+		cp /etc/resolv.conf /var/run/ppp/resolv.prev
+		rscf=/var/run/ppp/resolv.new
+		grep domain /var/run/ppp/resolv.prev > $rscf
+		grep search /var/run/ppp/resolv.prev >> $rscf
+		change_resolv_conf $rscf
+		rm -f $rscf
 	else
-		cp /etc/ppp/resolv.conf /etc
+		change_resolv_conf /var/run/ppp/resolv.conf
 	fi
 fi
 
