diff -up ppp-ppp-2.5.0/pppd/plugins/pppoatm/pppoatm.c.5~ ppp-ppp-2.5.0/pppd/plugins/pppoatm/pppoatm.c
--- ppp-ppp-2.5.0/pppd/plugins/pppoatm/pppoatm.c.5~	2023-03-18 09:11:05.000000000 +0000
+++ ppp-ppp-2.5.0/pppd/plugins/pppoatm/pppoatm.c	2024-07-14 00:44:26.323052442 +0100
@@ -146,7 +146,7 @@ static int connect_pppoatm(void)
 
 	if (!device_got_set)
 		no_device_given_pppoatm();
-	fd = socket(AF_ATMPVC, SOCK_DGRAM, 0);
+	fd = socket(AF_ATMPVC, SOCK_DGRAM | SOCK_CLOEXEC, 0);
 	if (fd < 0)
 		fatal("failed to create socket: %m");
 	memset(&qos, 0, sizeof qos);
diff -up ppp-ppp-2.5.0/pppd/plugins/pppoe/if.c.5~ ppp-ppp-2.5.0/pppd/plugins/pppoe/if.c
--- ppp-ppp-2.5.0/pppd/plugins/pppoe/if.c.5~	2023-03-18 09:11:05.000000000 +0000
+++ ppp-ppp-2.5.0/pppd/plugins/pppoe/if.c	2024-07-14 00:44:26.323052442 +0100
@@ -116,7 +116,7 @@ openInterface(char const *ifname, UINT16
     stype = SOCK_PACKET;
 #endif
 
-    if ((fd = socket(domain, stype, htons(type))) < 0) {
+    if ((fd = socket(domain, stype | SOCK_CLOEXEC, htons(type))) < 0) {
 	/* Give a more helpful message for the common error case */
 	if (errno == EPERM) {
 	    fatal("Cannot create raw socket -- pppoe must be run as root.");
diff -up ppp-ppp-2.5.0/pppd/plugins/pppoe/plugin.c.5~ ppp-ppp-2.5.0/pppd/plugins/pppoe/plugin.c
--- ppp-ppp-2.5.0/pppd/plugins/pppoe/plugin.c.5~	2023-03-18 09:11:05.000000000 +0000
+++ ppp-ppp-2.5.0/pppd/plugins/pppoe/plugin.c	2024-07-14 00:48:22.183488130 +0100
@@ -155,7 +155,7 @@ PPPOEConnectDevice(void)
     /* server equipment).                                                  */
     /* Opening this socket just before waitForPADS in the discovery()      */
     /* function would be more appropriate, but it would mess-up the code   */
-    conn->sessionSocket = socket(AF_PPPOX, SOCK_STREAM, PX_PROTO_OE);
+    conn->sessionSocket = socket(AF_PPPOX, SOCK_STREAM | SOCK_CLOEXEC, PX_PROTO_OE);
     if (conn->sessionSocket < 0) {
 	error("Failed to create PPPoE socket: %m");
 	return -1;
@@ -166,7 +166,7 @@ PPPOEConnectDevice(void)
     lcp_wantoptions[0].mru = conn->mru = conn->storedmru;
 
     /* Update maximum MRU */
-    s = socket(AF_INET, SOCK_DGRAM, 0);
+    s = socket(AF_INET, SOCK_DGRAM | SOCK_CLOEXEC, 0);
     if (s < 0) {
 	error("Can't get MTU for %s: %m", conn->ifName);
 	goto errout;
@@ -364,7 +364,7 @@ PPPoEDevnameHook(char *cmd, char **argv,
     }
 
     /* Open a socket */
-    if ((fd = socket(PF_PACKET, SOCK_RAW, 0)) < 0) {
+    if ((fd = socket(PF_PACKET, SOCK_RAW | SOCK_CLOEXEC, 0)) < 0) {
 	r = 0;
     }
 
diff -up ppp-ppp-2.5.0/pppd/plugins/pppoe/pppoe-discovery.c.5~ ppp-ppp-2.5.0/pppd/plugins/pppoe/pppoe-discovery.c
diff -up ppp-ppp-2.5.0/pppd/plugins/pppol2tp/openl2tp.c.5~ ppp-ppp-2.5.0/pppd/plugins/pppol2tp/openl2tp.c
--- ppp-ppp-2.5.0/pppd/plugins/pppol2tp/openl2tp.c.5~	2023-03-18 09:11:05.000000000 +0000
+++ ppp-ppp-2.5.0/pppd/plugins/pppol2tp/openl2tp.c	2024-07-14 00:44:26.323052442 +0100
@@ -93,7 +93,7 @@ static int openl2tp_client_create(void)
 	int result;
 
 	if (openl2tp_fd < 0) {
-		openl2tp_fd = socket(PF_UNIX, SOCK_DGRAM, 0);
+		openl2tp_fd = socket(PF_UNIX, SOCK_DGRAM | SOCK_CLOEXEC, 0);
 		if (openl2tp_fd < 0) {
 			error("openl2tp connection create: %m");
 			return -ENOTCONN;
diff -up ppp-ppp-2.5.0/pppd/plugins/pppol2tp/pppol2tp.c.5~ ppp-ppp-2.5.0/pppd/plugins/pppol2tp/pppol2tp.c
--- ppp-ppp-2.5.0/pppd/plugins/pppol2tp/pppol2tp.c.5~	2024-07-14 00:44:26.323052442 +0100
+++ ppp-ppp-2.5.0/pppd/plugins/pppol2tp/pppol2tp.c	2024-07-14 00:48:01.533447044 +0100
@@ -220,7 +220,7 @@ static void send_config_pppol2tp(int mtu
 		struct ifreq ifr;
 		int fd;
 
-		fd = socket(AF_INET, SOCK_DGRAM, 0);
+		fd = socket(AF_INET, SOCK_DGRAM | SOCK_CLOEXEC, 0);
 		if (fd >= 0) {
 			memset (&ifr, '\0', sizeof (ifr));
 			ppp_get_ifname(ifr.ifr_name, sizeof(ifr.ifr_name));
diff -up ppp-ppp-2.5.0/pppd/sys-linux.c.5~ ppp-ppp-2.5.0/pppd/sys-linux.c
--- ppp-ppp-2.5.0/pppd/sys-linux.c.5~	2024-07-14 00:44:26.293052392 +0100
+++ ppp-ppp-2.5.0/pppd/sys-linux.c	2024-07-14 00:49:20.623606918 +0100
@@ -499,12 +499,12 @@ static int modify_flags(int fd, int clea
 void sys_init(void)
 {
     /* Get an internet socket for doing socket ioctls. */
-    sock_fd = socket(AF_INET, SOCK_DGRAM, 0);
+    sock_fd = socket(AF_INET, SOCK_DGRAM | SOCK_CLOEXEC, 0);
     if (sock_fd < 0)
 	fatal("Couldn't create IP socket: %m(%d)", errno);
 
 #ifdef PPP_WITH_IPV6CP
-    sock6_fd = socket(AF_INET6, SOCK_DGRAM, 0);
+    sock6_fd = socket(AF_INET6, SOCK_DGRAM | SOCK_CLOEXEC, 0);
     if (sock6_fd < 0)
 	sock6_fd = -errno;	/* save errno for later */
 #endif
@@ -2675,7 +2675,7 @@ get_if_hwaddr(u_char *addr, char *name)
 	struct ifreq ifreq;
 	int ret, sock_fd;
 
-	sock_fd = socket(AF_INET, SOCK_DGRAM, 0);
+	sock_fd = socket(AF_INET, SOCK_DGRAM | SOCK_CLOEXEC, 0);
 	if (sock_fd < 0)
 		return -1;
 	memset(&ifreq.ifr_hwaddr, 0, sizeof(struct sockaddr));
@@ -2915,7 +2915,7 @@ int ppp_check_kernel_support(void)
 /*
  * Open a socket for doing the ioctl operations.
  */
-    s = socket(AF_INET, SOCK_DGRAM, 0);
+    s = socket(AF_INET, SOCK_DGRAM | SOCK_CLOEXEC, 0);
     if (s < 0)
 	return 0;
 
diff -up ppp-ppp-2.5.0/pppd/tty.c.5~ ppp-ppp-2.5.0/pppd/tty.c
--- ppp-ppp-2.5.0/pppd/tty.c.5~	2024-07-14 00:44:26.293052392 +0100
+++ ppp-ppp-2.5.0/pppd/tty.c	2024-07-14 00:44:26.323052442 +0100
@@ -942,7 +942,7 @@ open_socket(char *dest)
     *sep = ':';
 
     /* get a socket and connect it to the other end */
-    sock = socket(PF_INET, SOCK_STREAM, 0);
+    sock = socket(PF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0);
     if (sock < 0) {
 	error("Can't create socket: %m");
 	return -1;
