--- ppp-2.4.6/pppd/plugins/radius/Makefile.linux.radius~	2014-05-09 06:09:13.552468657 +0200
+++ ppp-2.4.6/pppd/plugins/radius/Makefile.linux	2014-05-09 06:10:25.902291356 +0200
@@ -39,17 +39,20 @@ install: all
 	$(INSTALL) -c -m 755 radius.so $(LIBDIR)
 	$(INSTALL) -c -m 755 radattr.so $(LIBDIR)
 	$(INSTALL) -c -m 755 radrealms.so $(LIBDIR)
-	$(INSTALL) -c -m 444 pppd-radius.8 $(MANDIR)
-	$(INSTALL) -c -m 444 pppd-radattr.8 $(MANDIR)
+	$(INSTALL) -c -m 644 pppd-radius.8 $(MANDIR)
+	$(INSTALL) -c -m 644 pppd-radattr.8 $(MANDIR)
 
 radius.so: radius.o libradiusclient.a
-	$(CC) -o radius.so -shared radius.o libradiusclient.a
+	$(CC) $(CFLAGS) -o radius.so -shared radius.o libradiusclient.a
 
 radattr.so: radattr.o
-	$(CC) -o radattr.so -shared radattr.o
+	$(CC) $(CFLAGS) -o radattr.so -shared radattr.o
 
 radrealms.so: radrealms.o
-	$(CC) -o radrealms.so -shared radrealms.o
+	$(CC) $(CFLAGS) -o radrealms.so -shared radrealms.o
+
+%.o: %.c
+	$(CC) $(CFLAGS) -c -o $@ $<
 
 CLIENTOBJS = avpair.o buildreq.o config.o dict.o ip_util.o \
 	clientid.o sendserver.o lock.o util.o md5.o
