--- ppp-2.4.6/pppd/main.c.1023~	2014-05-09 04:00:31.791693814 +0200
+++ ppp-2.4.6/pppd/main.c	2014-05-09 04:00:31.879691161 +0200
@@ -75,6 +75,7 @@
 #include <unistd.h>
 #include <signal.h>
 #include <errno.h>
+#define	__USE_GNU
 #include <fcntl.h>
 #include <syslog.h>
 #include <netdb.h>
--- ppp-2.4.6/pppd/options.c.1023~	2014-05-09 04:00:31.848692096 +0200
+++ ppp-2.4.6/pppd/options.c	2014-05-09 04:00:31.881691101 +0200
@@ -46,6 +46,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
+#define	__USE_GNU
 #include <fcntl.h>
 #include <stdlib.h>
 #include <syslog.h>
--- ppp-2.4.6/pppd/sys-linux.c.1023~	2014-05-09 04:00:31.859691764 +0200
+++ ppp-2.4.6/pppd/sys-linux.c	2014-05-09 04:02:38.624871206 +0200
@@ -74,6 +74,9 @@
 #include <sys/socket.h>
 #include <sys/time.h>
 #include <sys/errno.h>
+#ifdef __UCLIBC__
+#define	__USE_GNU
+#endif
 #include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/utsname.h>
--- ppp-2.4.6/pppd/tdb.c.1023~	2014-05-09 04:00:31.793693753 +0200
+++ ppp-2.4.6/pppd/tdb.c	2014-05-09 04:00:31.882691071 +0200
@@ -48,6 +48,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#define	__USE_GNU
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
--- ppp-2.4.6/pppd/tty.c.1023~	2014-05-09 04:00:31.797693633 +0200
+++ ppp-2.4.6/pppd/tty.c	2014-05-09 04:00:31.882691071 +0200
@@ -78,6 +78,7 @@
 #include <unistd.h>
 #include <signal.h>
 #include <errno.h>
+#define	__USE_GNU
 #include <fcntl.h>
 #include <syslog.h>
 #include <netdb.h>
--- ppp-2.4.6/pppd/utils.c.1023~	2014-05-09 04:00:31.824692819 +0200
+++ ppp-2.4.6/pppd/utils.c	2014-05-09 04:00:31.883691041 +0200
@@ -37,6 +37,7 @@
 #include <unistd.h>
 #include <signal.h>
 #include <errno.h>
+#define	__USE_GNU
 #include <fcntl.h>
 #include <syslog.h>
 #include <netdb.h>
