--- ppp-2.4.6/chat/Makefile.linux.1008~	2014-05-09 03:11:19.527730144 +0200
+++ ppp-2.4.6/chat/Makefile.linux	2014-05-09 03:11:19.612727578 +0200
@@ -18,10 +18,10 @@ INSTALL= install
 all:	chat
 
 chat:	chat.o
-	$(CC) -o chat chat.o
+	$(CC) -pie -o chat chat.o
 
 chat.o:	chat.c
-	$(CC) -c $(CFLAGS) -o chat.o chat.c
+	$(CC) -c $(CFLAGS) -fPIC -o chat.o chat.c
 
 install: chat
 	mkdir -p $(BINDIR) $(MANDIR)
--- ppp-2.4.6/pppd/Makefile.linux.1008~	2014-05-09 03:11:19.605727789 +0200
+++ ppp-2.4.6/pppd/Makefile.linux	2014-05-09 03:12:00.343496774 +0200
@@ -235,7 +235,7 @@ install: pppd
 	$(INSTALL) -c -m 644 pppd.8 $(MANDIR)
 
 pppd: $(PPPDOBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
+	$(CC) $(CFLAGS) -pie $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
 
 srp-entry:	srp-entry.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ srp-entry.c $(LIBS)
@@ -249,3 +249,6 @@ clean:
 
 depend:
 	$(CPP) -M $(CFLAGS) $(PPPDSRCS) >.depend
+
+%.o: %.c
+	 $(CC) -c $(CFLAGS) -fPIC -o $@ $<
