diff -up ppp-2.4.9/pppd/eap-tls.c.omv~ ppp-2.4.9/pppd/eap-tls.c
--- ppp-2.4.9/pppd/eap-tls.c.omv~	2021-01-11 13:57:45.264871267 +0100
+++ ppp-2.4.9/pppd/eap-tls.c	2021-01-11 13:59:10.781052518 +0100
@@ -278,7 +278,14 @@ ENGINE *eaptls_ssl_load_engine( char *en
     return e;
 }
 
-
+static int writer (UI *ui, UI_STRING *uis)
+{
+    PW_CB_DATA* cb_data = (PW_CB_DATA*)UI_get0_user_data(ui);
+    UI_set_result(ui, uis, cb_data->password);
+    return 1;
+};
+static int stub (UI* ui) {return 1;};
+static int stub_reader (UI *ui, UI_STRING *uis) {return 1;};
 
 /*
  * Initialize the SSL stacks and tests if certificates, key and crl
@@ -516,15 +523,6 @@ SSL_CTX *eaptls_init_ssl(int init_server
         {
             UI_METHOD* transfer_pin = UI_create_method("transfer_pin");
 
-            int writer (UI *ui, UI_STRING *uis)
-            {
-                PW_CB_DATA* cb_data = (PW_CB_DATA*)UI_get0_user_data(ui);
-                UI_set_result(ui, uis, cb_data->password);
-                return 1;
-            };
-            int stub (UI* ui) {return 1;};
-            int stub_reader (UI *ui, UI_STRING *uis) {return 1;};
-
             UI_method_set_writer(transfer_pin,  writer);
             UI_method_set_opener(transfer_pin,  stub);
             UI_method_set_closer(transfer_pin,  stub);
