--- ppp-2.4.6/pppd/plugins/dhcp/Makefile.linux.dhcp~	2014-05-09 11:15:13.394726132 +0200
+++ ppp-2.4.6/pppd/plugins/dhcp/Makefile.linux	2014-05-09 11:15:54.415869956 +0200
@@ -1,31 +1,34 @@
+DESTDIR = $(INSTROOT)/usr
+BINDIR = $(DESTDIR)/sbin
+MANDIR = $(DESTDIR)/share/man/man8
+LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(PPPDVERSION)
 
-PLUGIN=dhcpc.so
+PPPDVERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
+
+INSTALL	= install
+
+COPTS=$(RPM_OPT_FLAGS)
 CFLAGS=-I../.. -I../../../include -Wall $(RPM_OPT_FLAGS)
 
 SRCS = dhcpc.c clientpacket.c packet.c socket.c options.c
-OBJS = $(SRCS:%.c=%.lo)
-LIBS = -shared -Wl,-soname,dhcpc.so
-LTLIB = libdhcpc.la
+OBJS = $(SRCS:%.c=%.o)
 
-all: $(PLUGIN)
+all: dhcpc.so
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -s -c -m 755 dhcpc.so $(LIBDIR)
-
-$(PLUGIN): $(LTLIB)
-	gcc -o $@ $(OBJS:%.lo=.libs/%.o) $(LIBS)
+	$(INSTALL) -c -m 755 dhcpc.so $(LIBDIR)
 
-$(LTLIB): $(OBJS)
-	libtool --tag=CC --mode=link gcc $(CFLAGS) -o $@ $(OBJS) $(LIBS)
+dhcpc.so: $(OBJS)
+	$(CC) $(CFLAGS) -shared -Wl,-soname,dhcpc.so $^ -o $@
 
-%.lo: %.c
-	libtool --tag=CC --mode=compile gcc $(CFLAGS) -c $<
+%.o: %.c
+	$(CC) $(CFLAGS) -c -o $@ $<
 
 clean:
-	rm -f *.o *.lo *.so
+	rm -f *.o *.so
 
 distclean:
-	rm -f *.o *.lo *.so
+	rm -f *.o *.so
 
 dist-clean: distclean
