diff -up ppp-2.4.9/pppd/Makefile.linux.omv~ ppp-2.4.9/pppd/Makefile.linux
--- ppp-2.4.9/pppd/Makefile.linux.omv~	2021-01-11 14:09:18.274225410 +0100
+++ ppp-2.4.9/pppd/Makefile.linux	2021-01-11 14:09:24.491307798 +0100
@@ -156,7 +156,7 @@ endif
 
 ifdef NEEDDES
 ifndef USE_CRYPT
-CFLAGS   += -I$(shell $(CC) --print-sysroot)/usr/include/openssl
+CFLAGS   += -I/usr/include/openssl
 NEEDCRYPTOLIB = y
 else
 CFLAGS   += -DUSE_CRYPT=1
